src/utility/Auth.js

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useContext, useState, useEffect } from 'react'
+import firebase from 'firebase/app'
+import 'firebase/auth'
+
+const AuthContext = createContext()
+
+export function useAuth() {
+  return useContext(AuthContext)
+}
+
+export function AuthProvider({ children }) {
+  const [user, setUser] = useState(null)
+
+  useEffect(() => {
+    return firebase.auth().onAuthStateChanged(async currUser => {
+      setUser(currUser)
+    })
+  })
+
+  return (
+    <AuthContext.Provider value={{ isAuthed: !!user, user, setUser }}>
+      {children}
+    </AuthContext.Provider>
+  )
+}
+
+export const googleSignIn = async (setUser, setLocation) => {
+  await firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL)
+  const provider = new firebase.auth.GoogleAuthProvider()
+  try {
+    await firebase.auth().signInWithPopup(provider)
+    const user = firebase.auth().currentUser
+    setUser(user)
+    setLocation('/application')
```

Comment:
seeing as judging will also require users to be signed in, how do we want to handle this redirect? should we pass in a `redirTo` prop?

Reply:
Judging will require more nuanced authentication, so it should probably be a seperate login function

Reply:
what'll be the difference between application auth and registration auth?

Reply:
Oh, i see what you mean. Actually they will be the same, but how will we tell whether someone is going to judging vs application?

Reply:
I was thinking we can default to application. The only time they'll need to auth to judging is if they're already on livesite without being authed and click judging. In this case, we can probably use a parameter like `?redir=judging` or something and parse that to figure out where to redirect them

Reply:
Wouter doesn't have support for search params. I could implement it myself but it sounds a bit like scope creep. Maybe we can have a main landing page after login?

Reply:
or choose an already made page to be that landing page

Reply:
hmm looks like we'll do it like this: https://github.com/nwplus/livesite/pull/150#discussion_r532374941